packages/shylu/shylu_node/fastilu/src/shylu_fastilu.hpp

```diff
@@ -687,8 +688,10 @@ class FastILUPrec
               ExecSpace().fence();
             }
             //Sort each row of A by ColIdx
-            KokkosKernels::sort_crs_matrix<ExecSpace, OrdinalArray, OrdinalArray, ScalarArray>(aRowMapIn, aColIdxIn, aValIn);
-            ExecSpace().fence();
+            if (!skipSortMatrix) {
+              KokkosKernels::sort_crs_matrix<ExecSpace, OrdinalArray, OrdinalArray, ScalarArray>(aRowMapIn, aColIdxIn, aValIn);
```

Comment:
You could also skip sorting the matrix in symbolic above, around line 532.

Reply:
Never mind, that is not sorting the input A but rather the A with fill-in, so I think that one can't be skipped.

This sort also can't be skipped if you're applying the Metis column permutation to ``aColIdxIn``, even if ``skipSortMatrix == true``.

Reply:
@brian-kelley. Good point. I was checking this in Ifpack2, but will move it into FastILU. I'll also look at applying Metis ordering in Ifpack2 (as you suggested, but might be in another PR) so that we can skip this sort with Metis also..